options_test.go

```diff
@@ -0,0 +1,459 @@
+		if !c.expectedErr(err) {
+			t.Errorf("did not expect err: %s", err)
+		}
+	}
+}
+
+func TestHandleFifos(t *testing.T) {
+	validateTrue := func(options) bool { return true }
+	cases := []struct {
+		description  string
+		opt          options
+		outWriterNil bool
+		expectedErr  func(error) (bool, error)
+		numClosers   int
+		validate     func(options) bool
+		// TODO: Add a function to evaluate the ending state of the options
+		// since we are modifying it
+	}{
+		// Set both FcFifoLogfile and fclogfifo
+		// should be invalid
+		{
+			description: "Set both FcFifoLogFile and FcLogFifo",
+			opt: options{
+				FcFifoLogFile: "a",
+				FcLogFifo:     "b",
+			},
+			outWriterNil: true,
+			expectedErr: func(e error) (bool, error) {
+				return e == conflictingOptsSet, conflictingOptsSet
+			},
+			numClosers: 0,
+			validate:   validateTrue,
+		},
+
+		// Set fcfifologfile and have option that makes createFifoFileLogs fail
+		// should fail with proscribed error
+		{
+			description: "Set FcFifoLogFile causing createFifoFileLogs to fail",
+			opt: options{
+				FcFifoLogFile: "/noaccess",
+			},
+			outWriterNil: true,
+			expectedErr: func(a error) (bool, error) {
+				if a == nil {
+					return false, errors.New("failed to create fifo log file")
+				}
+				return strings.HasPrefix(a.Error(), "failed to create fifo log file"), errors.New("failed to create fifo log file")
+			},
+			numClosers: 0,
+			validate:   validateTrue,
+		},
+		// Set fclogfifo but not fcmetricsfifo
+		// fc metrics fifo should be created
+		{
+			description: "Set FcLogFifo but not FcMetricsFifo",
+			opt: options{
+				FcLogFifo: "testing",
+			},
+			outWriterNil: true,
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			numClosers: 1,
+			validate: func(opt options) bool {
+				return strings.HasSuffix(opt.FcMetricsFifo, "fc_metrics_fifo")
+			},
+		},
+		// set fcmetricsfifo but not fclogfifo
+		// fclogfifo should be created
+		{
+			description: "Set FcMetricsFifo but not FcLogFifo",
+			opt: options{
+				FcMetricsFifo: "test",
+			},
+			outWriterNil: true,
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			numClosers: 1,
+			validate: func(opt options) bool {
+				return strings.HasSuffix(opt.FcLogFifo, "fc_fifo")
+			},
+		},
+		// set fcfifologfile with valid value
+		// both fclogfifo and fcmetrics fifo should be created
+		{
+			description: "Set FcFifoLogFile with valid value",
+			opt: options{
+				FcFifoLogFile: "value",
+			},
+			outWriterNil: false,
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			numClosers: 2,
+			validate: func(opt options) bool {
+				return strings.HasSuffix(opt.FcLogFifo, "fc_fifo") &&
+					strings.HasSuffix(opt.FcMetricsFifo, "fc_metrics_fifo")
+			},
+		},
+	}
+	for _, c := range cases {
+		w, e := c.opt.handleFifos()
+		if (w == nil && !c.outWriterNil) || (w != nil && c.outWriterNil) {
+			t.Log("test:", c.description)
```

Comment:
Rather than logging how about adding `name string` to your cases structure.
```go
for _, c := range cases {
    t.Run(c.name, func(t *testing.T) {
       // Test here
    })
}
```

That way when it fails you know what test had failed without needing to log.